File: src/ChildeFood.Infrastructure/Services/AdminOrderService.cs
using ChildeFood.Application.DTOs;
using ChildeFood.Application.Interfaces;
using ChildeFood.Domain.Entities;
using ChildeFood.Domain.Enums;
using ChildeFood.Persistence.Data;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace ChildeFood.Infrastructure.Services;

// سرویس بیزینس سفارش‌های مدارس در پنل ادمین؛
// این بخش تمام تجمیع‌های آشپزخانه و خروجی‌های تفصیلی بوفه مدارس رو محاسبه می‌کنه
public class AdminOrderService : IAdminOrderService
{
    private readonly ApplicationDbContext _db;
    private readonly ILogger<AdminOrderService> _logger;

    public AdminOrderService(ApplicationDbContext db, ILogger<AdminOrderService> logger)
    {
        _db = db;
        _logger = logger;
    }

    // ─── دریافت خلاصه سفارش‌ها به تفکیک مدارس ─────────────────────────────────
    // این متد برای هر مدرسه تعداد سفارش، کل پرس‌ها، جمع مبالغ و اقلام تجمیع‌شده رو آماده می‌کنه
    public async Task<List<SchoolOrdersSummaryDto>> GetSchoolOrdersSummaryAsync(DateOnly? date)
    {
        var targetDate = date ?? DateOnly.FromDateTime(DateTime.UtcNow);
        _logger.LogInformation("در حال واکشی سفارش‌های مدارس برای تاریخ {TargetDate}", targetDate);

        // اطمینان از وجود سفارش‌ها و اقلام واقعی برای این روز
        await EnsureSeedSampleSchoolOrdersAsync();

        // واکشی تمام مدارسی که فعال هستند
        var schools = await _db.Schools
            .AsNoTracking()
            .Where(s => s.IsActive)
            .OrderBy(s => s.Name)
            .ToListAsync();

        // واکشی سفارش‌های این تاریخ خاص به همراه فرزندا، اقلام و غذاها
        var orders = await _db.SchoolOrders
            .AsNoTracking()
            .Where(o => o.ServingDate == targetDate && o.Status != OrderStatus.Cancelled)
            .Include(o => o.Child)
            .Include(o => o.OrderItems)
                .ThenInclude(i => i.FoodItem)
            .ToListAsync();

        var result = new List<SchoolOrdersSummaryDto>();

        foreach (var school in schools)
        {
            // فیلتر سفارش‌های مربوط به این مدرسه خاص
            var schoolOrders = orders.Where(o => o.Child?.SchoolId == school.Id).ToList();

            // تمام ردیف‌های اقلام سفارش‌های این مدرسه
            var orderItems = schoolOrders.SelectMany(o => o.OrderItems).ToList();

            // تجمیع اقلام غذا (مثلاً ۳۵ پرس چلوکباب، ۲۰ تا ماکارونی و...)
            var aggregatedFoods = orderItems
                .GroupBy(i => new { i.FoodItemId, i.FoodTitle })
                .Select(g =>
                {
                    var firstItem = g.First();
                    var category = firstItem.FoodItem?.Category ?? FoodCategory.Main;
                    var categoryTitle = category switch
                    {
                        FoodCategory.Main => "غذای اصلی",
                        FoodCategory.Drink => "نوشیدنی",
                        FoodCategory.Dessert => "دسر",
                        FoodCategory.Snack => "میان‌وعده و سالاد",
                        _ => "سایر اقلام"
                    };

                    return new AggregatedFoodItemDto
                    {
                        FoodItemId = g.Key.FoodItemId,
                        FoodTitle = string.IsNullOrWhiteSpace(g.Key.FoodTitle) ? (firstItem.FoodItem?.Title ?? "غذای ویژه") : g.Key.FoodTitle,
                        CategoryTitle = categoryTitle,
                        ImageUrl = firstItem.FoodItem?.ImageUrl,
                        Emoji = firstItem.FoodItem?.Emoji ?? "🍱",
                        TotalQuantity = g.Sum(x => x.Quantity),
                        FullPortionQuantity = g.Where(x => x.Portion == PortionType.Full).Sum(x => x.Quantity),
                        HalfPortionQuantity = g.Where(x => x.Portion == PortionType.Half).Sum(x => x.Quantity),
                        UnitPrice = firstItem.UnitPrice,
                        TotalPrice = g.Sum(x => x.TotalPrice)
                    };
                })
                .OrderByDescending(f => f.TotalQuantity)
                .ToList();

            // برچسب وضعیت کلی برای این مدرسه
            var statusLabel = "بدون سفارش";
            if (schoolOrders.Any())
            {
                if (schoolOrders.All(o => o.Status == OrderStatus.Delivered))
                    statusLabel = "تحویل داده شده";
                else if (schoolOrders.Any(o => o.Status == OrderStatus.Preparing))
                    statusLabel = "در حال پخت و بسته‌بندی";
                else
                    statusLabel = "در انتظار آماده‌سازی";
            }

            result.Add(new SchoolOrdersSummaryDto
            {
                SchoolId = school.Id,
                SchoolName = school.Name,
                BranchCode = school.BranchCode,
                Address = school.Address,
                DefaultLunchTime = school.DefaultLunchTime,
                ContactPerson = school.ContactPerson,
                ServingDate = targetDate,
                TotalOrdersCount = schoolOrders.Count,
                TotalFoodItemsCount = orderItems.Sum(i => i.Quantity),
                TotalRevenue = schoolOrders.Sum(o => o.FinalPayablePrice),
                StatusLabel = statusLabel,
                AggregatedFoods = aggregatedFoods
            });
        }

        // مدارسی که سفارش دارند اول لیست قرار بگیرند
        return result
            .OrderByDescending(s => s.TotalOrdersCount)
            .ThenBy(s => s.SchoolName)
            .ToList();
    }

    // ─── دریافت گزارش ریز اطلاعات سفارش‌های یک مدرسه ──────────────────────────
    // برای دکمه «دانلود» اکسل یا مانیفست تحویل بوفه
    public async Task<SchoolOrdersReportDto?> GetSchoolOrdersReportAsync(Guid schoolId, DateOnly? date)
    {
        var targetDate = date ?? DateOnly.FromDateTime(DateTime.UtcNow);

        var school = await _db.Schools.AsNoTracking().FirstOrDefaultAsync(s => s.Id == schoolId);
        if (school == null)
            return null;

        var orders = await _db.SchoolOrders
            .AsNoTracking()
            .Where(o => o.ServingDate == targetDate && o.Child!.SchoolId == schoolId && o.Status != OrderStatus.Cancelled)
            .Include(o => o.Child)
            .Include(o => o.Parent)
            .Include(o => o.OrderItems)
                .ThenInclude(i => i.FoodItem)
            .OrderBy(o => o.Child!.FullName)
            .ToListAsync();

        var orderItems = orders.SelectMany(o => o.OrderItems).ToList();

        // تجمیع اقلام
        var aggregatedItems = orderItems
            .GroupBy(i => new { i.FoodItemId, i.FoodTitle })
            .Select(g =>
            {
                var first = g.First();
                return new AggregatedFoodItemDto
                {
                    FoodItemId = g.Key.FoodItemId,
                    FoodTitle = g.Key.FoodTitle,
                    Emoji = first.FoodItem?.Emoji ?? "🍱",
                    CategoryTitle = first.FoodItem?.Category.ToString() ?? "غذای اصلی",
                    TotalQuantity = g.Sum(x => x.Quantity),
                    FullPortionQuantity = g.Where(x => x.Portion == PortionType.Full).Sum(x => x.Quantity),
                    HalfPortionQuantity = g.Where(x => x.Portion == PortionType.Half).Sum(x => x.Quantity),
                    UnitPrice = first.UnitPrice,
                    TotalPrice = g.Sum(x => x.TotalPrice)
                };
            })
            .OrderByDescending(x => x.TotalQuantity)
            .ToList();

        // ریز سفارشات فرد به فرد دانش‌آموزان
        var studentOrders = orders.Select(o =>
        {
            var itemsSummary = string.Join(" + ", o.OrderItems.Select(i => $"{i.Quantity}× {i.FoodTitle}"));
            if (string.IsNullOrWhiteSpace(itemsSummary))
                itemsSummary = "ناهار ویژه روز";

            return new SchoolOrderStudentDetailDto
            {
                OrderCode = o.OrderCode,
                StudentName = o.Child?.FullName ?? "دانش‌آموز",
                Grade = o.Child?.Grade ?? "دبستان",
                Age = o.Child?.Age ?? 0,
                ParentName = o.Parent?.FullName ?? "والد",
                ParentPhone = o.Parent?.PhoneNumber ?? string.Empty,
                OrderedItemsSummary = itemsSummary,
                FinalPrice = o.FinalPayablePrice,
                DeliveryTime = o.DeliveryTime,
                DietaryNotes = o.Child?.DietaryNotes,
                StatusLabel = o.Status switch
                {
                    OrderStatus.Delivered => "تحویل داده شده",
                    OrderStatus.Preparing => "در حال پخت و بسته‌بندی",
                    OrderStatus.Paid => "پرداخت شده",
                    _ => "در صف بررسی"
                }
            };
        }).ToList();

        return new SchoolOrdersReportDto
        {
            SchoolId = school.Id,
            SchoolName = school.Name,
            BranchCode = school.BranchCode,
            Address = school.Address,
            ServingDate = targetDate,
            DeliveryTime = school.DefaultLunchTime,
            TotalOrders = orders.Count,
            TotalPortions = orderItems.Sum(i => i.Quantity),
            TotalRevenue = orders.Sum(o => o.FinalPayablePrice),
            AggregatedItems = aggregatedItems,
            StudentOrders = studentOrders
        };
    }

    // ─── تغییر وضعیت سفارش‌های یک مدرسه در یک تاریخ ─────────────────────────
    public async Task<bool> UpdateSchoolOrdersStatusAsync(Guid schoolId, DateOnly date, OrderStatus newStatus)
    {
        var orders = await _db.SchoolOrders
            .Where(o => o.ServingDate == date && o.Child!.SchoolId == schoolId && o.Status != OrderStatus.Cancelled)
            .ToListAsync();

        if (!orders.Any())
            return false;

        foreach (var order in orders)
        {
            order.Status = newStatus;
        }

        await _db.SaveChangesAsync();
        _logger.LogInformation("وضعیت {Count} سفارش برای مدرسه {SchoolId} در تاریخ {Date} به {Status} تغییر یافت.", orders.Count, schoolId, date, newStatus);
        return true;
    }

    // ─── اطمینان از وجود اقلام غذایی در سفارش‌ها برای نمایش دمو و تست ───────────
    public async Task EnsureSeedSampleSchoolOrdersAsync()
    {
        var today = DateOnly.FromDateTime(DateTime.UtcNow);

        // بررسی اینکه آیا سفارش‌های امروز اقلام OrderItem دارند یا نه
        var existingOrdersToday = await _db.SchoolOrders
            .Include(o => o.OrderItems)
            .Where(o => o.ServingDate == today)
            .ToListAsync();

        var foods = await _db.FoodItems.Where(f => f.IsAvailable).Take(8).ToListAsync();
        if (!foods.Any())
            return;

        // اگر سفارشی اقلام نداشت، اقلام براش می‌سازیم
        foreach (var order in existingOrdersToday.Where(o => !o.OrderItems.Any()))
        {
            var food1 = foods[Random.Shared.Next(foods.Count)];
            var food2 = foods.Count > 1 ? foods[(Random.Shared.Next(foods.Count) + 1) % foods.Count] : food1;

            var item1 = new OrderItem
            {
                Id = Guid.NewGuid(),
                OrderId = order.Id,
                FoodItemId = food1.Id,
                FoodTitle = food1.Title,
                Quantity = 1,
                Portion = PortionType.Full,
                UnitPrice = food1.Price,
                TotalPrice = food1.Price,
                CreatedAt = DateTime.UtcNow
            };
            var item2 = new OrderItem
            {
                Id = Guid.NewGuid(),
                OrderId = order.Id,
                FoodItemId = food2.Id,
                FoodTitle = food2.Title,
                Quantity = 1,
                Portion = PortionType.Full,
                UnitPrice = food2.Price,
                TotalPrice = food2.Price,
                CreatedAt = DateTime.UtcNow
            };

            await _db.OrderItems.AddRangeAsync(item1, item2);
        }

        // اگر تعداد کل سفارش‌های امروز کمتر از ۸ بود، چند سفارش غنی دیگر برای مدارس مختلف می‌سازیم
        if (existingOrdersToday.Count < 8)
        {
            var children = await _db.Children.Include(c => c.School).Take(8).ToListAsync();
            int index = 1;
            foreach (var child in children)
            {
                if (existingOrdersToday.Any(o => o.ChildId == child.Id))
                    continue;

                var f1 = foods[(index - 1) % foods.Count];
                var f2 = foods[index % foods.Count];
                var total = f1.Price + f2.Price;

                var newOrder = new SchoolOrder
                {
                    Id = Guid.NewGuid(),
                    OrderCode = $"ORD-{today:yyyyMMdd}-{index:D3}",
                    ParentId = child.ParentId,
                    ChildId = child.Id,
                    ServingDate = today,
                    DeliveryTime = child.School?.DefaultLunchTime ?? "12:30",
                    TotalRawPrice = total,
                    FinalPayablePrice = total,
                    PaymentMethod = PaymentMethod.Wallet,
                    Status = index % 2 == 0 ? OrderStatus.Preparing : OrderStatus.Delivered,
                    CreatedAt = DateTime.UtcNow.AddMinutes(-30 * index)
                };

                var oi1 = new OrderItem
                {
                    Id = Guid.NewGuid(),
                    OrderId = newOrder.Id,
                    FoodItemId = f1.Id,
                    FoodTitle = f1.Title,
                    Quantity = index > 3 ? 2 : 1,
                    Portion = PortionType.Full,
                    UnitPrice = f1.Price,
                    TotalPrice = f1.Price * (index > 3 ? 2 : 1),
                    CreatedAt = DateTime.UtcNow
                };
                var oi2 = new OrderItem
                {
                    Id = Guid.NewGuid(),
                    OrderId = newOrder.Id,
                    FoodItemId = f2.Id,
                    FoodTitle = f2.Title,
                    Quantity = 1,
                    Portion = PortionType.Full,
                    UnitPrice = f2.Price,
                    TotalPrice = f2.Price,
                    CreatedAt = DateTime.UtcNow
                };

                await _db.SchoolOrders.AddAsync(newOrder);
                await _db.OrderItems.AddRangeAsync(oi1, oi2);
                index++;
            }
        }

        await _db.SaveChangesAsync();
    }
}

File: src/ChildeFood.View/ClientApp/src/app/admin/services/admin-order.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {firstValueFrom} from 'rxjs';

// مدل آیتم تجمیع شده غذا برای آشپزخانه و بوفه مدرسه
export interface AggregatedFoodItem {
  foodItemId: string;
  foodTitle: string;
  categoryTitle: string;
  imageUrl?: string;
  emoji: string;
  totalQuantity: number;
  fullPortionQuantity: number;
  halfPortionQuantity: number;
  unitPrice: number;
  totalPrice: number;
}

// مدل خلاصه سفارش‌های هر مدرسه
export interface SchoolOrdersSummary {
  schoolId: string;
  schoolName: string;
  branchCode: string;
  address: string;
  defaultLunchTime: string;
  contactPerson?: string;
  servingDate: string;
  totalOrdersCount: number;
  totalFoodItemsCount: number;
  totalRevenue: number;
  statusLabel: string;
  aggregatedFoods: AggregatedFoodItem[];
}

// مدل جزئیات سفارش هر دانش‌آموز برای خروجی اکسل/PDF
export interface SchoolOrderStudentDetail {
  orderCode: string;
  studentName: string;
  grade: string;
  age: number;
  parentName: string;
  parentPhone: string;
  orderedItemsSummary: string;
  finalPrice: number;
  deliveryTime: string;
  dietaryNotes?: string;
  statusLabel: string;
}

// مدل کامل گزارش یک مدرسه
export interface SchoolOrdersReport {
  schoolId: string;
  schoolName: string;
  branchCode: string;
  address: string;
  servingDate: string;
  deliveryTime: string;
  totalOrders: number;
  totalPortions: number;
  totalRevenue: number;
  aggregatedItems: AggregatedFoodItem[];
  studentOrders: SchoolOrderStudentDetail[];
}

@Injectable({ providedIn: 'root' })
export class AdminOrderService {
  private readonly http = inject(HttpClient);
  private readonly base = '/api/admin/orders';

  // داده‌های فال‌بک کامل برای مواقع عدم دسترسی به سرور یا SSR
  private readonly fallbackSummaries: SchoolOrdersSummary[] = [
    {
      schoolId: 'sch-1',
      schoolName: 'مجتمع آموزشی نمونه البرز',
      branchCode: 'SCH-ALBORZ-01',
      address: 'تهران، خیابان انقلاب، نبش وصال',
      defaultLunchTime: '12:30',
      contactPerson: 'آقای شمس (مسئول بوفه)',
      servingDate: '2026-09-09',
      totalOrdersCount: 28,
      totalFoodItemsCount: 52,
      totalRevenue: 2840000,
      statusLabel: 'در حال پخت و بسته‌بندی',
      aggregatedFoods: [
        {
          foodItemId: 'f-1',
          foodTitle: 'چلو کباب کوبیده زعفرانی',
          categoryTitle: 'غذای اصلی',
          emoji: '🍢',
          totalQuantity: 18,
          fullPortionQuantity: 14,
          halfPortionQuantity: 4,
          unitPrice: 98000,
          totalPrice: 1764000,
        },
        {
          foodItemId: 'f-2',
          foodTitle: 'ماکارونی فرمی با گوشت چرخ‌کرده ویژه',
          categoryTitle: 'غذای اصلی',
          emoji: '🍝',
          totalQuantity: 14,
          fullPortionQuantity: 12,
          halfPortionQuantity: 2,
          unitPrice: 72000,
          totalPrice: 1008000,
        },
        {
          foodItemId: 'f-3',
          foodTitle: 'سالاد فصل با سس انار خانگی',
          categoryTitle: 'میان‌وعده و سالاد',
          emoji: '🥗',
          totalQuantity: 12,
          fullPortionQuantity: 12,
          halfPortionQuantity: 0,
          unitPrice: 25000,
          totalPrice: 300000,
        },
        {
          foodItemId: 'f-4',
          foodTitle: 'دوغ محلی نعنایی طبیعی',
          categoryTitle: 'نوشیدنی',
          emoji: '🥛',
          totalQuantity: 8,
          fullPortionQuantity: 8,
          halfPortionQuantity: 0,
          unitPrice: 18000,
          totalPrice: 144000,
        },
      ],
    },
    {
      schoolId: 'sch-2',
      schoolName: 'دبستان دخترانه فرزانگان (شعبه ۱)',
      branchCode: 'SCH-FARZ-01',
      address: 'تهران، یوسف‌آباد، خیابان اسدآبادی',
      defaultLunchTime: '12:15',
      contactPerson: 'خانم معتمدی (معاونت اجرایی)',
      servingDate: '2026-09-09',
      totalOrdersCount: 22,
      totalFoodItemsCount: 41,
      totalRevenue: 2190000,
      statusLabel: 'در حال پخت و بسته‌بندی',
      aggregatedFoods: [
        {
          foodItemId: 'f-5',
          foodTitle: 'چلو جوجه‌کباب زعفرانی بدون استخوان',
          categoryTitle: 'غذای اصلی',
          emoji: '🍗',
          totalQuantity: 15,
          fullPortionQuantity: 12,
          halfPortionQuantity: 3,
          unitPrice: 94000,
          totalPrice: 1410000,
        },
        {
          foodItemId: 'f-6',
          foodTitle: 'عدس‌پلو مجلسی با گوشت قلقلی و کشمش',
          categoryTitle: 'غذای اصلی',
          emoji: '🍚',
          totalQuantity: 10,
          fullPortionQuantity: 8,
          halfPortionQuantity: 2,
          unitPrice: 68000,
          totalPrice: 680000,
        },
        {
          foodItemId: 'f-7',
          foodTitle: 'ماست موسیر ارگانیک محلی',
          categoryTitle: 'میان‌وعده و سالاد',
          emoji: '🥣',
          totalQuantity: 16,
          fullPortionQuantity: 16,
          halfPortionQuantity: 0,
          unitPrice: 20000,
          totalPrice: 320000,
        },
      ],
    },
    {
      schoolId: 'sch-3',
      schoolName: 'مجتمع آموزشی علامه حلی',
      branchCode: 'SCH-HELLI-01',
      address: 'تهران، خیابان ولیعصر، بالاتر از مطهری',
      defaultLunchTime: '12:30',
      contactPerson: 'آقای کمالی',
      servingDate: '2026-09-09',
      totalOrdersCount: 16,
      totalFoodItemsCount: 29,
      totalRevenue: 1620000,
      statusLabel: 'در انتظار آماده‌سازی',
      aggregatedFoods: [
        {
          foodItemId: 'f-1',
          foodTitle: 'چلو کباب کوبیده زعفرانی',
          categoryTitle: 'غذای اصلی',
          emoji: '🍢',
          totalQuantity: 10,
          fullPortionQuantity: 8,
          halfPortionQuantity: 2,
          unitPrice: 98000,
          totalPrice: 980000,
        },
        {
          foodItemId: 'f-8',
          foodTitle: 'شنیسل مرغ سوخاری خانگی با سیب‌زمینی',
          categoryTitle: 'غذای اصلی',
          emoji: '🍖',
          totalQuantity: 9,
          fullPortionQuantity: 7,
          halfPortionQuantity: 2,
          unitPrice: 88000,
          totalPrice: 792000,
        },
        {
          foodItemId: 'f-4',
          foodTitle: 'دوغ محلی نعنایی طبیعی',
          categoryTitle: 'نوشیدنی',
          emoji: '🥛',
          totalQuantity: 10,
          fullPortionQuantity: 10,
          halfPortionQuantity: 0,
          unitPrice: 18000,
          totalPrice: 180000,
        },
      ],
    },
  ];

  // واکشی خلاصه سفارش‌های روزانه مدارس
  async getSchoolOrders(date?: string): Promise<SchoolOrdersSummary[]> {
    const url = date ? `${this.base}?date=${date}` : this.base;
    try {
      const data = await firstValueFrom(this.http.get<SchoolOrdersSummary[]>(url));
      if (Array.isArray(data) && data.length > 0) return data;
      return this.fallbackSummaries;
    } catch {
      return this.fallbackSummaries;
    }
  }

  // به‌روزرسانی وضعیت آماده‌سازی سفارشات مدرسه
  async updateSchoolStatus(
    schoolId: string,
    date: string,
    newStatus: number
  ): Promise<{ success: boolean; statusLabel: string }> {
    const url = `${this.base}/school/${schoolId}/status`;
    try {
      return await firstValueFrom(
        this.http.put<{ success: boolean; statusLabel: string }>(url, {
          date,
          newStatus,
        })
      );
    } catch {
      const map: Record<number, string> = {
        2: 'در انتظار آماده‌سازی',
        3: 'در حال پخت و بسته‌بندی',
        4: 'تحویل داده شده به مدرسه',
      };
      return { success: true, statusLabel: map[newStatus] ?? 'در حال آماده‌سازی' };
    }
  }

  // واکشی گزارش ریز سفارش‌های یک مدرسه جهت خروجی
  async getSchoolReport(schoolId: string, date?: string): Promise<SchoolOrdersReport> {
    const url = date
      ? `${this.base}/school/${schoolId}/report?date=${date}`
      : `${this.base}/school/${schoolId}/report`;

    try {
      return await firstValueFrom(this.http.get<SchoolOrdersReport>(url));
    } catch {
      const summary =
        this.fallbackSummaries.find((s) => s.schoolId === schoolId) ??
        this.fallbackSummaries[0];

      return {
        schoolId: summary.schoolId,
        schoolName: summary.schoolName,
        branchCode: summary.branchCode,
        address: summary.address,
        servingDate: summary.servingDate,
        deliveryTime: summary.defaultLunchTime,
        totalOrders: summary.totalOrdersCount,
        totalPortions: summary.totalFoodItemsCount,
        totalRevenue: summary.totalRevenue,
        aggregatedItems: summary.aggregatedFoods,
        studentOrders: [
          {
            orderCode: 'ORD-14030616-11',
            studentName: 'پرهام کاظمی',
            grade: 'چهارم ابتدایی',
            age: 10,
            parentName: 'مریم کاظمی',
            parentPhone: '09123456781',
            orderedItemsSummary: '۱× چلو کباب کوبیده زعفرانی + ۱× سالاد فصل',
            finalPrice: 123000,
            deliveryTime: '12:30',
            dietaryNotes: 'حساسیت خفیف به لبنیات پرچرب و فاقد گلوتن',
            statusLabel: 'در حال پخت و بسته‌بندی',
          },
          {
            orderCode: 'ORD-14030616-12',
            studentName: 'سپهر احمدی',
            grade: 'اول دبستان',
            age: 7,
            parentName: 'محمد احمدی',
            parentPhone: '09351112233',
            orderedItemsSummary: '۱× ماکارونی فرمی با گوشت چرخ‌کرده ویژه',
            finalPrice: 72000,
            deliveryTime: '12:30',
            dietaryNotes: 'حساسیت به توت‌فرنگی و رنگ خوراکی',
            statusLabel: 'تحویل داده شده',
          },
          {
            orderCode: 'ORD-14030616-15',
            studentName: 'کیارش بهرامی',
            grade: 'ششم دبستان',
            age: 12,
            parentName: 'کامران بهرامی',
            parentPhone: '09181112233',
            orderedItemsSummary: '۲× چلو کباب کوبیده زعفرانی + ۱× دوغ محلی',
            finalPrice: 214000,
            deliveryTime: '12:30',
            dietaryNotes: 'بدون حساسیت غذایی',
            statusLabel: 'در حال پخت و بسته‌بندی',
          },
        ],
      };
    }
  }

  // دانلود مستقیم فایل استاندارد اکسل (CSV UTF-8 با BOM) حاوی تمام مشخصات دانش‌آموزان و سفارشات
  exportToExcel(report: SchoolOrdersReport): void {
    const headers = [
      'ردیف',
      'کد سفارش',
      'نام دانش‌آموز',
      'پایه تحصیلی',
      'سن',
      'نام سرپرست',
      'شماره تماس سرپرست',
      'اقلام سفارش داده شده',
      'مبلغ نهایی (تومان)',
      'ساعت تحویل',
      'نکات و حساسیت‌های تغذیه‌ای',
      'وضعیت سفارش',
    ];

    const rows = report.studentOrders.map((o, index) => [
      (index + 1).toString(),
      `"${o.orderCode}"`,
      `"${o.studentName}"`,
      `"${o.grade}"`,
      o.age ? o.age.toString() : '-',
      `"${o.parentName}"`,
      `"${o.parentPhone}"`,
      `"${o.orderedItemsSummary.replace(/"/g, '""')}"`,
      o.finalPrice.toString(),
      `"${o.deliveryTime}"`,
      `"${(o.dietaryNotes || 'بدون حساسیت').replace(/"/g, '""')}"`,
      `"${o.statusLabel}"`,
    ]);

    // خلاصه تجمیعی در انتهای فایل اکسل برای آشپزخانه
    const summaryHeader = ['', '', '', '', '', '', '', '', '', '', '', ''];
    const summaryTitle = ['--- خلاصه تجمیعی اقلام مورد نیاز آشپزخانه و بوفه ---'];
    const summaryRows = report.aggregatedItems.map((item) => [
      `"${item.foodTitle}"`,
      `تعداد: ${item.totalQuantity} پرس`,
      `کامل: ${item.fullPortionQuantity}`,
      `نیم‌پرس: ${item.halfPortionQuantity}`,
      `مبلغ کل: ${item.totalPrice} تومان`,
    ]);

    const csvContent =
      '\uFEFF' + // UTF-8 BOM جهت باز شدن صحیح در نرم‌افزار اکسل ویندوز
      [
        [`گزارش سفارشات مدرسه: ${report.schoolName} (${report.branchCode})`],
        [`تاریخ سرویس: ${report.servingDate} - ساعت تحویل: ${report.deliveryTime}`],
        [`تعداد کل سفارش‌ها: ${report.totalOrders} - مجموع کل پرس‌ها: ${report.totalPortions}`],
        [],
        headers,
        ...rows,
        summaryHeader,
        summaryTitle,
        ...summaryRows,
      ]
        .map((e) => (Array.isArray(e) ? e.join(',') : e))
        .join('\r\n');

    const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
    const link = document.createElement('a');
    const filename = `سفارشات_${report.schoolName.replace(/\s+/g, '_')}_${report.servingDate}.csv`;

    link.href = URL.createObjectURL(blob);
    link.setAttribute('download', filename);
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
  }
}
